ca/ca.go

```diff
@@ -269,3 +272,26 @@ func (ca *CAImpl) GetRootCert() *core.Certificate {
 	}
 	return ca.root.cert
 }
+
+func (ca *CAImpl) GetRootKey() *rsa.PrivateKey {
```

Comment:
I would change this function and the corresponding `GetIntermediateKey` to return the interface type (`crypto.Signer`) and then use the type switch I mentioned earlier in the caller. That will leave the CA more general and make it easier to support different key types in the future.

Reply:
See above.